src/toxcore/dht/server.rs

```diff
@@ -839,38 +866,39 @@ mod tests {
         let nat_res = NatPingResponse { id: random_u64() };
         let nat_payload = DhtRequestPayload::NatPingResponse(nat_res);
         let dht_req = DhtRequest::new(&precomp, &bob_pk, &bob_pk, nat_payload.clone());
-        let client = alice.create_client(&addr, bob_pk);
-        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res).wait().is_ok());
+        let mut state = alice.state.write();
+        let client = alice.create_client(&addr, bob_pk, &state.peers_cache);
+        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res, &state.kbucket).wait().is_ok());
         // if receiver' pk != node's pk and receiver's pk exists in kbucket, returns ok()
         let pn = PackedNode::new(false, SocketAddr::V4("127.0.0.1:12345".parse().unwrap()), &bob_pk);
 
-        alice.try_add_to_kbucket(&pn);
+        alice.try_add_to_kbucket(&pn, &mut state.kbucket);
         let nat_res = NatPingResponse { id: random_u64() };
         let nat_payload = DhtRequestPayload::NatPingResponse(nat_res);
         let dht_req = DhtRequest::new(&precomp, &bob_pk, &bob_pk, nat_payload.clone());
-        let client = alice.create_client(&addr, bob_pk);
-        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res).wait().is_ok());
+        let client = alice.create_client(&addr, bob_pk, &state.peers_cache);
+        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res, &state.kbucket).wait().is_ok());
```

Comment:
You should test it by passing special packet and calling `handle_packet`, not by calling handle_YOUR_PACKET directly. See https://github.com/tox-rs/tox/blob/master/src/toxcore/tcp/server/server.rs#L430